apps/meteor/ee/app/livechat-enterprise/server/api/lib/tags.ts

```diff
@@ -27,9 +29,22 @@ type FindTagsByIdParams = {
 
 type FindTagsByIdResult = ILivechatTag | null;
 
-export async function findTags({ text, pagination: { offset, count, sort } }: FindTagsParams): Promise<FindTagsResult> {
+// If viewAll is true, then all tags will be returned, regardless of department
+// If viewAll is false, then all public tags will be returned, and
+//  if department is specified, then all department tags will be returned
+export async function findTags({
+	text,
+	department,
+	viewAll,
+	pagination: { offset, count, sort },
+}: FindTagsParams): Promise<FindTagsResult> {
 	const query = {
-		...(text && { $or: [{ name: new RegExp(escapeRegExp(text), 'i') }, { description: new RegExp(escapeRegExp(text), 'i') }] }),
+		...((text || !viewAll) && {
```

Comment:
This still allows users to see tags from other departments just by knowing the id 😬 we should add a check of user's departments and then return the tag list of _only_ the departments user can see

Reply:
Ouch! I missed that! We'd need to address this before merging this PR.

Reply:
![image](https://github.com/RocketChat/Rocket.Chat/assets/34130764/c69a3c52-f71c-4c8c-9b6a-7794bac7b95a)

It should be fixed now